### ISSUE TITLE
Query hangs connection when doing a transaction in another terminal

### ISSUE THREAD

**Issue by douglasselias** [cockroachdb/cockroach#61269]
**Describe the problem**

Doing a query in one terminal while in another terminal there is a transaction occurring, the query hangs.

**To Reproduce**

1. Open two terminals and execute this: `cockroach sql --insecure`
2. On the first terminal execute: `CREATE TABLE IF NOT EXISTS "users" ("id"   SERIAL , "name" VARCHAR(255), "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL, "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));`
3. On the second terminal execute: `START TRANSACTION;`
4. Still on the second terminal execute: `INSERT INTO "users" ("id","name","createdAt","updatedAt") VALUES (DEFAULT,'bob','2021-03-01 13:05:02.546 +00:00','2021-03-01 13:05:02.546 +00:00') RETURNING "id","name","createdAt","updatedAt";`
5. And finally, on the first terminal execute: `SELECT "id", "name", "createdAt", "updatedAt" FROM "users" AS "user";`

**Expected behavior**

Should return the result from the query.

**Environment:**
 - CockroachDB version: v20.2.4
 - OS: Pop!_OS 20.10 x86_64

**Comment by rafiss** [comment]
Might be something similar to https://github.com/cockroachdb/cockroach/issues/60754

**Comment by papb** [comment]
Interesting, this might be the cause for the "test crashes" that I experienced when [setting up tests in sequelize-cockroachdb](https://github.com/cockroachdb/sequelize-cockroachdb/pull/51). I ended up splitting the test execution in more than 100 parts. Once this issue here is fixed, I suggest trying again to have the test execute entirely in a single job, and see if it works.

**Comment by ajwerner** [comment]
Wait, what about this is an issue? In cockroach, writes which precede reads hold exclusive locks. This differs from Postgres but is certainly not wrong from any sort of correctness perspective. If you need the read to proceed and push the write, then you can set it to have a higher priority using SET TRANSACTION PRIORITY HIGH.

**Comment by rafiss** [comment]
@ajwerner I don't think this is a CockroachDB bug. For context, @douglasselias is a contractor working with the SQL Experience team on adding support for CockroachDB to sequelize, and @papb is a sequelize maintainer. I believe @RichardJCai asked for this issue to be filed in order to track findings and to figure out how we want to unblock this test or change the test itself.

Changing the test so that it sets the transaction priority sounds like a good approach. Are we able to modify the test to work differently for CockroachDB versus PostgreSQL?

If not that, can the test be refactored to run the transaction concurrently in a different thread?

**Comment by douglasselias** [comment]
I tried with `SET TRANSACTION PRIORITY HIGH` but still doesn't work.

Using a thread will also force the sequelize-cockroach users to use it as well.
Besides, changing the test would diverge from the sequelize integration test, so I think it would be better to document this limitation on the sequelize-cockroach readme.

For additional context, here is the [test](https://github.com/sequelize/sequelize/blob/main/test/integration/cls.test.js#L120-L137)

**Comment by RichardJCai** [comment]
> I tried with `SET TRANSACTION PRIORITY HIGH` but still doesn't work.
> 
> Using a thread will also force the sequelize-cockroach users to use it as well.
> Besides, changing the test would diverge from the sequelize integration test, so I think it would be better to document this limitation on the sequelize-cockroach readme.
> 
> For additional context, here is the [test](https://github.com/sequelize/sequelize/blob/main/test/integration/cls.test.js#L120-L137)

Seems reasonable, I think we can open an issue in the sequelize-cockroach repo and document it and close the issue here.

**Comment by andreimatei** [comment]
@richardartoul please link to the new issue and close this if there's nothing else

**Comment by RichardJCai** [comment]
> @richardartoul please link to the new issue and close this if there's nothing else

Wrong Richard to tag here :P 
Anyways opened issue in sequelize-cockroachdb package and closing this issue.

**Comment by papb** [comment]
> the query hangs.

So is it the correct behavior to hang forever? Wouldn't a timeout or deadlock or auto-rollback or something else be better?

I think hanging indefinitely might in particular be the cause of [the "test crashes"](https://github.com/cockroachdb/cockroach/issues/61269#issuecomment-789002026)...

**Comment by ajwerner** [comment]
It's not a deadlock if one of the connections can make progress. It seems like that connection is just choosing not to. If there were indeed a real deadlock (as in two transactions blocked on eachother inside the system) then it would indeed be broke. Cockroach has statement timeouts by they are opt-in with the `statement_timeout` session variable.

**Comment by papb** [comment]
Oops, indeed, it's not a deadlock. I stand corrected.

> Cockroach has statement timeouts by they are opt-in with the statement_timeout session variable.

Cool!
